Dynamic taxness for price display (SHOOP-2166)

suutari-ai opened the pull request:
See the commit messages.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1638/
Test PASSed.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1639/
Test PASSed.

tulimaki commented on shoop/core/models/_contacts.py:
```diff
@@ -91,6 +113,12 @@ def __init__(self, *args, **kwargs):
         super(Contact, self).__init__(*args, **kwargs)
 
+    def get_price_display_options(self):
+        group_with_options = self.groups.with_price_display_options().first()
```
I think first() is not correct here. IMO if contact has multiple contact groups set with pricing options, then default group should determine the options.

suutari-ai replied:
Sounds feasible. I will implement that.

tulimaki commented:
I think the ``PriceDisplayOptions`` is not a good naming here. It's more like a price display mode and the options is to show_prices_including_taxes and hide_prices. Then based on these two options actual pricing mode is set.

tulimaki commented:
Can you add these new ``ContactGroup`` fields into admin also so these can be actually set from admin? I thnk the ``ContactGroupBaseForm`` is good place for these.

suutari-ai commented:
> Can you add these new ContactGroup fields into admin also so these can be actually set from admin? I thnk the ContactGroupBaseForm is good place for these.

That's not in scope of SHOOP-2166, but rather SHOOP-2143, so let's do that in another PR. (Yeah, I know, this PR already has some parts of SHOOP-2143: the Classic Gray changes and contact group additions, but that's only because it was hard to check usability of these new filters without trying them with the templates...)

suutari-ai commented:
> I think the PriceDisplayOptions is not a good naming here. It's more like a price display mode and the options is to show_prices_including_taxes and hide_prices. Then based on these two options actual pricing mode is set.

I disagree. PriceDisplayMode sounds more like an enumeration, but the class is a collection of options rather than enumeration. Although currently there is just four different ways to show prices ((1) don't show them, (2) show them as taxful, (3) show them as taxless, (4) show them with original taxness), it might very well be that in the future PriceDisplayOptions class may have things like `taxless_price_template = '{price} (ALV 0%)'`, or `hide_base_prices = True  # to show discounted prices as if they were not discounted`, etc. I stand with the naming.

tulimaki commented:
You should add some tests for these if possible. At least ``shoop/core/utils/prices.py`` should have tests.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1646/
Test FAILed.

suutari-ai commented:
PR updated. Changes since Friday:
* Updated `Contact.get_price_display_options` as requested.
* Added tests.
* Moved PriceDisplayOptions to shoop.core.pricing.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1647/
Test FAILed.

suutari-ai commented:
retest this please

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1648/
Test PASSed.

tulimaki commented:
@suutari-ai thank you so much!
